src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -63,7 +61,9 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor, ILoggerFactory
                 _options.Clock = new SystemClock();
             }
 
-            _lastExpirationScan = _options.Clock.UtcNow;
+            _timer = new System.Timers.Timer(Math.Max(1, _options.ExpirationScanFrequency.TotalMilliseconds));
+            _timer.Elapsed += OnTimerElapsed;
+            _timer.Start();
```

Comment:
This is bad for testability. Any timing mechanism needs to be abstracted for the unit tests.

Reply:
I agree, but it's still doable